docs/web/tutorial-common-lifecycle-hooks.mdx

```diff
@@ -0,0 +1,156 @@
+---
+title: Lifecycle Hooks for Starting Workflows
+description: VertiGIS Studio Web - Learn what lifecycle hooks are available and when to use them.
+---
+
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+# Lifecycle Hooks for Starting Workflows
+
+This article will tell you about common lifecycle hooks within Web and when you should use them for starting Workflows.
+
+# Services > App > Events >
+
+## Application Initializing
+
+Event gets executed when Web begins to initialize. Commands or Workflows used here must not contain blocking actions such as alerts or user-facing forms; these will prevent the application from loading. This can be useful for launching Workflows that are going to impact the initial state of the application (such as initial layout).
+
+## Application Initialized
+
+This event gets executed when Web has initialized. Commands or Workflows used here must not contain blocking actions such as alerts or user-facing forms; these will prevent the application from loading in Designer.
+
+# Components > Map > Events >
+
+## Map Initialized
+
+This is an event that gets triggered when the map is initialized. The map is initialized when it is loaded for the first time, a user signs in or out, or in Mobile when a map area is activated or deactivated. This can be useful for displaying a Workflow that presents UI such as a form or for a Workflow that depends upon map data.
+
+# Services > Launch Link Workflows
+
+Using the Launch Link service you can add and remove Workflows to the app that can be optionally specified and run by a URL parameter. The launch link Workflow is run at the same time as the Map Initialized Event. This can be useful for passing in information such as an asset id and then using a Workflow to have Web start with that asset already centered and selected.
+
+<img
+    src={useBaseUrl("img/web-workflow-lifecycle-launch-link-3.png")}
+/>
+
+You can generate these URLs via Web Designer's App Launch Link Dialog. (Deploy > Development > 3 Dot Menu > Get Links)
+
+<img
+    src={useBaseUrl("img/web-workflow-lifecycle-launch-link-1.png")}
+/>
+
+Use the generated link to run the specified workflow at startup.
+
+<img
+    src={useBaseUrl("img/web-workflow-lifecycle-launch-link-2.png")}
+/>
+
+# Event Service
+
+You can configure a Workflow to run on any VertiGIS Studio Web [Event](https://developers.vertigisstudio.com/docs/Web/api-commands-operations-events/#events) by registering an event listener in the [Event service](https://developers.vertigisstudio.com/blog/2023/03/01/mobile-generic-event-listener/).
+
+:::danger Warning: In general configuring an event listener for a Workflow that contains alerts or forms can be undesirable, special consideration should also be given to the performance of a Workflow that is triggered by an Event that is likely to run often as this could degrade the user experience.
```

Comment:
```suggestion
:::danger Warning: In general configuring an event listener for a Workflow that contains alerts or forms can be undesirable. Special consideration should also be given to the performance of a Workflow that is triggered by an Event that is likely to run often, as this could degrade the user experience.
```